packages/protocol/src/action-protocol/base-protocol.ts

```diff
@@ -13,114 +13,224 @@
+     */
+    export function hasKind(object: any, kind: string): object is Action {
+        return RequestAction.is(object) && object.kind === kind;
+    }
+
+    export function generateRequestId(): string {
+        return sprotty.generateRequestId();
+    }
 }
+
 /**
  * A response action is sent to respond to a request action. The responseId must match the requestId of the preceding request.
- * In case the responseId is empty or undefined, the action is handled as standalone, i.e. it was fired without a preceding request.
+ * In case the responseId is empty, the action is handled as standalone, i.e. it was fired without a preceding request.
+ * The action dispatcher of the GLSP server has a special handling for {@link RequestAction} handlers
+ * and automatically sets the `responseId` of the corresponding responseAction. So on the server side its typically enough
+ * to set the `responseId` to an empty string and rely on the `ActionDispatcher` for assigning the correct `responseId.
+ * Additional typeguard functions are provided via the corresponding namespace.
  */
-export interface ResponseAction extends Action {
+export interface ResponseAction extends Action, sprotty.ResponseAction {
     /**
      * Id corresponding to the request this action responds to.
      */
-    readonly responseId: string;
+    responseId: string;
 }
 
-export function isResponseAction(action?: any): action is ResponseAction {
-    return isAction(action) && 'responseId' in action && typeof action['responseId'] === 'string' && action['responseId'] !== '';
+export namespace ResponseAction {
+    export function is(object: any): object is ResponseAction {
+        return Action.is(object) && hasStringProp(object, 'responseId');
+    }
+
+    /**
+     * Typeguard function to check wether the given object is an {@link ResponseAction} with a non-empty response id.
+     * @param object The object to check.
+     * @returns A type literal indicating wether the given object is a response action with a non-empty response id.
+     */
+    export function hasValidResponseId(object: any): object is ResponseAction {
+        return ResponseAction.is(object) && object.responseId !== '';
+    }
 }
 
 /**
- * A reject action is a response fired to indicate that a request must be rejected.
+ * A reject action is a {@link ResponseAction} fired to indicate that a certain a certain {@link ResponseAction}
+ * has been rejected.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `RejectActions`.
  */
-export class RejectAction implements ResponseAction {
-    static readonly KIND = 'rejectRequest';
-    readonly kind = RejectAction.KIND;
-
-    constructor(public readonly message: string, public readonly responseId: string, public readonly detail?: JsonAny) {}
+export interface RejectAction extends ResponseAction, sprotty.RejectAction {
+    /**
+     * The unique action kind.
+     */
```

Comment:
I'd spare such comments, also if it hurts to be "incomplete". But imho they just add noise and add no information.